apps/frontend/src/pages/quiz/libs/components/onboarding/onboarding.tsx

```diff
@@ -0,0 +1,147 @@
+import { Button, Loader, ProgressBar } from "~/libs/components/components.js";
+import { PREVIOUS_INDEX_OFFSET } from "~/libs/constants/constants.js";
+import {
+	useAppDispatch,
+	useAppForm,
+	useAppSelector,
+	useCallback,
+	useEffect,
+} from "~/libs/hooks/hooks.js";
+import { actions as onboardingActions } from "~/modules/onboarding/onboarding.js";
+
+import { OnboardingAnswer } from "./libs/components/components.js";
+import { ONBOARDING_FORM_DEFAULT_VALUES } from "./libs/constants/constants.js";
+import { type OnboardingFormValues } from "./libs/types/types.js";
+import styles from "./styles.module.css";
+
+type Properties = {
+	onNext: () => void;
+};
+
+type FormToSave = {
+	[key: string]: string;
+};
+
+const Onboarding: React.FC<Properties> = ({ onNext }: Properties) => {
+	const dispatch = useAppDispatch();
+	const {
+		currentQuestionIndex,
+		isLastQuestion,
+		question,
+		totalQuestionsAmount,
+		userId,
+	} = useAppSelector(({ auth, onboarding }) => ({
+		currentQuestionIndex: onboarding.currentQuestionIndex,
+		isLastQuestion:
+			onboarding.currentQuestionIndex ===
+			onboarding.questions.length - PREVIOUS_INDEX_OFFSET,
+		question: onboarding.currentQuestion,
+		totalQuestionsAmount: onboarding.questions.length,
+		userId: auth.user?.id,
+	}));
+
+	useEffect(() => {
+		void dispatch(onboardingActions.getAll());
+	}, [dispatch]);
+
+	const { control, handleSubmit, isValid } = useAppForm<OnboardingFormValues>({
+		defaultValues: ONBOARDING_FORM_DEFAULT_VALUES,
+	});
+
+	const getAnswerIds = useCallback((formData: FormToSave) => {
+		return Object.values(formData).map(Number);
+	}, []);
+
+	const handleNextStep = useCallback(
+		(data: OnboardingFormValues) => {
+			if (question && data[`answer${question.id.toString()}`] !== undefined) {
```

Comment:
```suggestion
			if (question && hasAnswer) {
```
Move to a separate variable